Jenkinsfile.buildchain

```
@@ -0,0 +1,111 @@
+@Library('jenkins-pipeline-shared-libraries')_
+
+additionalTimeout = "${env.ADDITIONAL_TIMEOUT?.trim() ?: 120}"
+
+agentLabel = "${env.ADDITIONAL_LABEL?.trim() ? ADDITIONAL_LABEL : 'kie-rhel7 && kie-mem16g'} && !master"
+build_jdk_tool = env.BUILD_JDK_TOOL?.trim() ?: 'kie-jdk11'
+build_maven_tool = env.BUILD_MAVEN_TOOL?.trim() ?: 'kie-maven-3.8.1'
+
+pr_type = env.BUILDCHAIN_PR_TYPE?.trim() ?: 'pr'
+build_chain_additional_arguments = env.BUILDCHAIN_PROJECT ? "-sp=${env.BUILDCHAIN_PROJECT?.trim()}" : ''
```

why do you need the `?.` in `env.BUILDCHAIN_PROJECT?.trim()` as you already check it is not empty ?

---

@radtriste I'm trying to assign a default `pr_type` in case the `BUILDCHAIN_PR_TYPE` env variable is not set and at the same time I want to trim the `BUILDCHAIN_PR_TYPE` value. Is it ok for you?

---

good for me

---

and for my previous commit, why don't you just do:
```suggestion
build_chain_additional_arguments = env.BUILDCHAIN_PROJECT ? "-sp=${env.BUILDCHAIN_PROJECT.trim()}" : ''
```
I know it is nitpicking but just to know if I am missing something :)

---

you are right, the null value is double checked. You proposal is better